examples/Stockagent/agent.py

```
@@ -0,0 +1,427 @@
+
+        loan_format_check, fail_response, loan = self.secretary.check_loan(resp,
+                                                                           max_loan)  # secretary check loan format
+        while not loan_format_check:
+            # log.logger.debug("WARNING: Loan format check failed because of these issues: {}".format(fail_response))
+            try_times += 1
+            if try_times > MAX_TRY_TIMES:
+                log.logger.warning("WARNING: Loan format try times > MAX_TRY_TIMES. Skip as no loan today.")
+                loan = {"loan": "no"}
+                break
+
+            resp = self.run_api(format_prompt(LOAN_RETRY_PROMPT, {"fail_response": fail_response}))
```

Ruff (0.14.3)

`LOAN_RETRY_PROMPT` may be undefined, or defined from star imports

(F405)